libcontainer/cgroups/utils.go

```diff
@@ -34,8 +35,14 @@ var (
 func IsCgroup2UnifiedMode() bool {
 	isUnifiedOnce.Do(func() {
 		var st unix.Statfs_t
-		if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
-			panic("cannot statfs cgroup root")
+		err := unix.Statfs(unifiedMountpoint, &st)
+		if err != nil && os.IsNotExist(err) && system.RunningInUserNS() {
+			// /sys/fs/cgroup not found, likely rootless running in user NS
+			logrus.WithError(err).Debugf("%s missing, assuming cgroup v1", unifiedMountpoint)
+			isUnified = false
+			return
+		} else if err != nil {
+			panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
```

Comment:
Can I suggest changing the error handling structure slightly, for better readability? Something like

```suggestion
		if err != nil {
		  if os.IsNotExist(err) && system.RunningInUserNS() {
		    // ignore the "not found" error if running in userns
		  	logrus.WithError(err).Debugf("%s missing, assuming cgroup v1", unifiedMountpoint)
  			isUnified = false
  			return
      }
      panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
```

Reply:
Yeah, thanks, that is better.  Updated.